### ISSUE TITLE
"ipdb" should be swapped with "pdbpp" (pdb++) in dependencies

### ISSUE THREAD

**Issue by mdmintz** [seleniumbase/SeleniumBase#1615]
### "ipdb" should be swapped with "pdbpp" (pdb++) in dependencies

The ``ipdb`` dependency has too many sub dependencies:

```bash
pip install ipdb

Collecting ipdb
  Using cached ipdb-0.13.9-py3-none-any.whl
Requirement already satisfied: setuptools in /Users/michael/.virtualenvs/ipdb/lib/python3.11/site-packages (from ipdb) (65.5.0)
Collecting ipython>=7.17.0
  Using cached ipython-8.6.0-py3-none-any.whl (761 kB)
Collecting toml>=0.10.2
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting decorator
  Using cached decorator-5.1.1-py3-none-any.whl (9.1 kB)
Collecting backcall
  Using cached backcall-0.2.0-py2.py3-none-any.whl (11 kB)
Collecting jedi>=0.16
  Using cached jedi-0.18.1-py2.py3-none-any.whl (1.6 MB)
Collecting matplotlib-inline
  Using cached matplotlib_inline-0.1.6-py3-none-any.whl (9.4 kB)
Collecting pickleshare
  Using cached pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)
Collecting prompt-toolkit<3.1.0,>3.0.1
  Using cached prompt_toolkit-3.0.33-py3-none-any.whl (383 kB)
Collecting pygments>=2.4.0
  Using cached Pygments-2.13.0-py3-none-any.whl (1.1 MB)
Collecting stack-data
  Using cached stack_data-0.6.1-py3-none-any.whl (24 kB)
Collecting traitlets>=5
  Using cached traitlets-5.5.0-py3-none-any.whl (107 kB)
Collecting pexpect>4.3
  Using cached pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
Collecting appnope
  Using cached appnope-0.1.3-py2.py3-none-any.whl (4.4 kB)
Collecting parso<0.9.0,>=0.8.0
  Using cached parso-0.8.3-py2.py3-none-any.whl (100 kB)
Collecting ptyprocess>=0.5
  Using cached ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)
Collecting wcwidth
  Using cached wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
Collecting executing>=1.2.0
  Using cached executing-1.2.0-py2.py3-none-any.whl (24 kB)
Collecting asttokens>=2.1.0
  Using cached asttokens-2.1.0-py2.py3-none-any.whl (26 kB)
Collecting pure-eval
  Using cached pure_eval-0.2.2-py3-none-any.whl (11 kB)
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)

Installing collected packages:
wcwidth,
pure-eval,
ptyprocess,
pickleshare,
executing,
backcall,
appnope,
traitlets,
toml,
six,
pygments,
prompt-toolkit,
pexpect,
parso,
decorator,
matplotlib-inline,
jedi,
asttokens,
stack-data,
ipython,
ipdb
```

Wow! That might be close to 40% of all SeleniumBase dependencies!

The good news is that there may be a much lighter alternative: ``pdbpp`` (pdb++)

```bash
pip install pdbpp

Collecting pdbpp
  Using cached pdbpp-0.10.3-py2.py3-none-any.whl (23 kB)
Collecting fancycompleter>=0.8
  Using cached fancycompleter-0.9.1-py3-none-any.whl (9.7 kB)
Collecting wmctrl
  Using cached wmctrl-0.4-py3-none-any.whl
Collecting pygments
  Using cached Pygments-2.13.0-py3-none-any.whl (1.1 MB)
Collecting pyrepl>=0.8.2
  Using cached pyrepl-0.9.0-py3-none-any.whl

Installing collected packages:
wmctrl,
pyrepl,
pygments,
fancycompleter,
pdbpp
```

Looks like ``pdbpp`` is a lot lighter than ``ipdb``!

Maybe it's time to swap out ``ipdb`` (https://github.com/gotcha/ipdb) with ``pdbpp`` (https://github.com/pdbpp/pdbpp).

**Comment by mdmintz** [comment]
ADDITIONAL NOTES:
* Although ``ipdb`` will get removed from included dependencies as a result of this change, people can still install it separately if they like it.
* There appears to be a bug in ``ipdb`` on Python 3.11 that leads to a Python segmentation fault when in a breakpoint and performing a "n" (next) command when the next line is a Selenium command. (Seen on an M1/M2 Mac)
* There is an overlapping dependency between ``ipdb`` and ``pdbpp``: ``pygments``, so that one will stay after the switch. Some of the other ``ipdb`` dependencies are also used by other packages, (such as ``six``), so those dependencies will stay as well.

**Comment by mdmintz** [comment]
After this change has been made, anyone with an existing ``pytest.ini`` file will need to REMOVE THE FOLLOWING PART to avoid issues after uninstalling ``ipdb`` / ``IPython``:

``--pdbcls=IPython.terminal.debugger:TerminalPdb``

**Comment by mdmintz** [comment]
More details on the ``ipdb`` / ``IPython`` "Segmentation fault":

```bash
Fatal Python error: Segmentation fault

Thread 0x00000001728bb000 (most recent call first):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/concurrent/futures/thread.py", line 81 in _worker
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 975 in run
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1038 in _bootstrap_inner
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 995 in _bootstrap

Thread 0x00000001718af000 (most recent call first):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 320 in wait
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 622 in wait
  File "/Users/michael/.virtualenvs/sbase11b/lib/python3.11/site-packages/IPython/core/history.py", line 881 in run
  File "/Users/michael/.virtualenvs/sbase11b/lib/python3.11/site-packages/IPython/core/history.py", line 60 in only_when_enabled
  File "/Users/michael/.virtualenvs/sbase11b/lib/python3.11/site-packages/decorator.py", line 232 in fun
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1038 in _bootstrap_inner
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 995 in _bootstrap
```

This is actually the main reason I want to switch away from``ipdb``. But there are plenty of other bonuses to do so.

**Comment by mdmintz** [comment]
Released in ``4.9.0`` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.9.0

**Comment by mdmintz** [comment]
For anyone following this thread, I made changes again, swapping ``pdbpp`` with my own ``pdbp``. See details here: https://github.com/seleniumbase/SeleniumBase/issues/1623

The GitHub repo for ``pdbp`` is here: https://github.com/mdmintz/pdbp

In summary, ``pdbp`` fixes the issues that were in ``pdbpp``. ``pdbp`` also simplifies things (by reducing dependencies) and adds new features. Here's everything that gets installed when installing ``pdbp`` from scratch on POSIX systems such as macOS and Linux: (NOTE: The ``pyreadline3`` dependency will also be installed on Windows systems.)

```bash
pip install pdbp

Collecting pdbp
  Using cached pdbp-1.2.5-py2.py3-none-any.whl (14 kB)
Collecting tabcompleter>=1.1.0
  Using cached tabcompleter-1.1.0-py2.py3-none-any.whl (6.7 kB)
Collecting six>=1.16.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting pygments>=2.13.0
  Using cached Pygments-2.13.0-py3-none-any.whl (1.1 MB)

Installing collected packages:
tabcompleter,
six,
pygments,
pdbp

Successfully installed pdbp-1.2.5 pygments-2.13.0 six-1.16.0 tabcompleter-1.1.0
```

As seen above, this is a massive reduction in the complexity seen from installing ``ipdb``.
